pkg/e2e/up_test.go

```diff
@@ -153,3 +153,16 @@ func TestScaleDoesntRecreate(t *testing.T) {
 	assert.Check(t, !strings.Contains(res.Combined(), "Recreated"))
 
 }
+
+func TestUpWithDependencyNotRequired(t *testing.T) {
+	c := NewCLI(t)
+	const projectName = "compose-e2e-dependency-not-required"
+	t.Cleanup(func() {
+		c.RunDockerComposeCmd(t, "--project-name", projectName, "down")
+	})
+
+	res := c.RunDockerComposeCmd(t, "-f", "./fixtures/dependencies/deps-not-required.yaml", "--project-name", projectName,
+		"up", "-d")
+	assert.Assert(t, strings.Contains(res.Combined(), "foo"), res.Combined())
+	assert.Assert(t, !strings.Contains(res.Combined(), "bar"), res.Combined())
```

Comment:
From compose-spec/compose-spec#382:
> Compose only warns you when the dependency service isn't started or available.

```
❯ docker compose up
[+] Running 2/0
 ✔ Network required_default  Created                                                                                                                  0.0s
 ✔ Container required-foo-1  Created                                                                                                                  0.0s
Attaching to required-foo-1
required-foo-1  | foo
required-foo-1 exited with code 0
```

Shouldn't there be a warning "`foo` has optional dependency on `bar` which was not selected by active profiles" or something?

Reply:
Right!